Bugsnag/Assets/Bugsnag/Editor/SymbolUpload/BugsnagAndroidSymbolUploader.cs

```diff
@@ -0,0 +1,246 @@
+                EditorUtility.ClearProgressBar();
+                return;
+            }
+
+            // Check if a cli path was provided in the settings object, if yes use it if not download the cli
+            string cliExecutablePath;
+            if (!string.IsNullOrEmpty(config.BugsnagCLIExecutablePath))
+            {
+                cliExecutablePath = config.BugsnagCLIExecutablePath;
+            }
+            else
+            {
+                if (!DownloadBugsnagCli(GetBugsnagCliDownloadUrl(), _cliDownloadPath) || !MakeExecutable(_cliDownloadPath))
+                {
+                    UnityEngine.Debug.LogError("Failed to download and make the Bugsnag CLI executable.");
+                    EditorUtility.ClearProgressBar();
+                    return; // Exit if the download or chmod process failed
+                }
+                cliExecutablePath = _cliDownloadPath;
+            }
+
+
+            var buildOutputPath = Path.GetDirectoryName(report.summary.outputPath);
+
+            var args = string.Format("upload unity-android --api-key={0} --verbose --project-root={1} {2}", apiKey, Application.dataPath, buildOutputPath);
+            if (!string.IsNullOrEmpty(config.AppVersion))
+            {
+                args += $" --version-name={config.AppVersion}";
+            }
+            if (config.VersionCode > -1)
+            {
+                args += $" --version-code={config.VersionCode}";
+            }
+            RunBugsnagCliCommand(cliExecutablePath, args);
+            EditorUtility.ClearProgressBar();
+        }
+
+        private string GetBugsnagCliDownloadUrl()
+        {
+
+            // Detect platform
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return RuntimeInformation.OSArchitecture == Architecture.Arm64
+                    ? _cliDownloadUrl + "arm64-macos-bugsnag-cli"
+                    : _cliDownloadUrl + "x86_64-macos-bugsnag-cli";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return RuntimeInformation.OSArchitecture == Architecture.X86
+                    ? _cliDownloadUrl + "i386-windows-bugsnag-cli.exe"
+                    : _cliDownloadUrl + "x86_64-windows-bugsnag-cli.exe";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return RuntimeInformation.OSArchitecture == Architecture.X86
+                    ? _cliDownloadUrl + "i386-linux-bugsnag-cli"
+                    : _cliDownloadUrl + "x86_64-linux-bugsnag-cli";
+            }
+
+            throw new PlatformNotSupportedException("Your platform is not supported for Bugsnag CLI.");
+        }
+
+        private bool DownloadBugsnagCli(string cliUrl, string downloadPath)
+        {
+            UnityEngine.Debug.Log("Starting Bugsnag CLI download process...");
```

Comment:
This seems misleading as the download may not even happen. I suggest removing this message and also the "already exists" one on L133 and only log a message if the download is required given it'll only happen on rare occasions, you can safely assume the absence of a "downloading" message means that it didn't.